### DISCUSSION TITLE
Confused about double-sided relationships

### DISCUSSION THREAD

**Discussion by YummYume** [pocketbase/pocketbase#2356] [Q&A] [answered]
Hello, thank you for making PocketBase :clap: 

Coming from very classic ORMs, I am used to having double-sided relationships.
I am confused however about how PocketBase handles that currently.

Let's say I have a list with items, and I've added a `items` field to my list.
From what I understand, I can expand my queries to get the full list of my `items` instead of just ids. Seems good.
I can also get the current list my item is attached to by expanding it with something like `lists(item)`. Seems good too.

But when it comes to inserting, updating & deleting relations I am pretty confused.
Is it possible to force having only one list per item? To have a true ManyToOne relationship? Can I change the list that the item is linked to without explicitly creating a new relation field?
Can I also insert new records directly into my list and not have to create them one by one? This is something I find very useful and time-saving with ORMs.

For example, in js :

```ts
await pb.collection('lists').create({
  ...,
  items: [
    { name: 'item1' },
    ...
  ]
})
```

Thank you for your time.

**Comment by ganigeorgiev** [comment, marked as answer]
> Is it possible to force having only one list per item? To have a true ManyToOne relationship? Can I change the list that the item is linked to without explicitly creating a new relation field?

I don't understand the questions. Could you elaborate a little more and/or provide an example with sample data and the expected workflow?

---

> Can I also insert new records directly into my list and not have to create them one by one?

No, this is not supported at the moment. You have to create the record(s) upfront and pass their `id`(s).

**Reply by ganigeorgiev** [reply]
> Is it possible to force having only one list per item?

If "items" in your example is a multiple `relation` field, you may want to inverse the relation, aka. have a single `relation` "list" field pointing to the lists collection.

**Reply by ganigeorgiev** [reply]
_For a similar todo-list type of data structure example, you can also check https://github.com/pocketbase/pocketbase/discussions/1776._

**Reply by YummYume** [reply]
Thank you @ganigeorgiev for such a quick answer. Effectively, it makes sense to have my relation moved to the `items` collection in this case.
As for the provided example, it does answer my question on how I would need to handle nested relations with PocketBase :+1:

My only concern now is how can I keep track effectively of each item on my list. For example, whenever my form is submitted, one item might have been deleted, another one updated, and two others created... Not to mention items will have other items themselves, but I'm keeping it simple for the example. This is most likely my job to check for this after the form submission, but I'm wondering if it will become a problem to possibly make 20+ requests per save to save each item individually?

**Reply by ganigeorgiev** [reply]
>  For example, whenever my form is submitted, one item might have been deleted, another one updated, and two others created.

If the relation is inversed (_aka. there is only a single `relation` field "list" in the items collection_), then you don't really have to track/sync manually the list.

If you want to reload a list record on item create/update/delete, you can subscribe to the `items` collections. The only drawback of this is that currently the realtime subscriptions doesn't support `?filter/sort/expand` parameters and you'll have to manually filter the event item inside the callback:

```js
const list = ...

pb.collection("items").subscribe("*", (e) => {
    if (e.record.list != list.id) {
        return; // not an item from the current list
    }

    // trigger list reload OR manually check the item action and apply
    // the necessary list transformation
    if (e.action == "create") {
        ...
    }
});
```

> if it will become a problem to possibly make 20+ requests per save to save each item individually

The server can handle it. Bulk transactional create/update/delete will be also available as part of #48 sometime in the future.

**Reply by YummYume** [reply]
@ganigeorgiev Thanks for the clarification. I will follow that issue closely then, it's really something that I'd want.

For now, I'll have to figure out myself if an item was just added, updated or deleted. That really shouldn't be that hard so it's fine.
